packages/harpguru-core/src/components/menu-close-button/menu-close-button.tsx

```diff
@@ -1,15 +1,36 @@
-import { TapGestureHandler } from 'react-native-gesture-handler'
+import { useTimingTransition } from 'react-native-redash'
+import Animated, { Easing, interpolate } from 'react-native-reanimated'
+import { TapGestureHandler, State } from 'react-native-gesture-handler'
+import type { TapGestureHandlerStateChangeEvent } from 'react-native-gesture-handler'
 import { StyleSheet, View } from 'react-native'
 import React from 'react'
 import { AntDesign } from '@expo/vector-icons'
 
 import type { MenuProps } from '../../types'
 import { getSizes, colors } from '../../styles'
+import { tapAnimationDuration } from '../../constants'
 
 export const MenuCloseButton = ({
   openCloseTapHandler,
 }: Pick<MenuProps, 'openCloseTapHandler'>): React.ReactElement => {
   const sizes = getSizes()
+  const [isTapped, setIsTapped] = React.useState(false)
```

Comment:
It would be good if a single custom hook could be used to manage `isTapped` status and the associated animation values, and possibly the even handlers too. There are exceptions; the HarpCell returns different states depending on whether they've been tapped on or tapped off, but I'd like to use a similar, perhaps less sophisticated model elsewhere too.